apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/notice.js

```diff
@@ -1,98 +1,124 @@
 /* global wpcomGlobalStyles */
-import { ExternalLink, Notice } from '@wordpress/components';
-import { useSelect } from '@wordpress/data';
-import { createInterpolateElement, render, useEffect, useState } from '@wordpress/element';
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useEffect, useCallback } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
-import './notice.scss';
-import { recordUpgradePreSaveNoticeClick, recordUpgradeNoticePreSaveShow } from './tracks-events';
 import { useGlobalStylesConfig } from './use-global-styles-config';
+import './notice.scss';
 
-function GlobalStylesNoticeComponent() {
-	const { siteChanges, globalStylesInUse } = useGlobalStylesConfig();
+const NOTICE_ID = 'wpcom-global-styles/gating-notice';
 
-	// Closes the sidebar if there are no more changes to be saved.
-	useEffect( () => {
-		if ( ! siteChanges.length ) {
-			/*
-			 * This uses a fragile CSS selector to target the cancel button which might be broken on
-			 * future releases of Gutenberg. Unfortunately, Gutenberg doesn't provide any mechanism
-			 * for closing the sidebar – everything is handled using an internal state that it is not
-			 * exposed publicly.
-			 *
-			 * See https://github.com/WordPress/gutenberg/blob/0b30a4cb34d39c9627b6a3795a18aee21019ce25/packages/edit-site/src/components/editor/index.js#L137-L138.
-			 */
-			document.querySelector( '.entities-saved-states__panel-header button:last-child' )?.click();
-		}
-	}, [ siteChanges ] );
+const GlobalStylesNotice = () => {
+	const { globalStylesInUse, globalStylesId } = useGlobalStylesConfig();
+	const { isSiteEditor, isPostEditor } = useSelect(
+		( select ) => ( {
+			isSiteEditor: !! select( 'core/edit-site' ),
+			isPostEditor: ! select( 'core/edit-site' ) && !! select( 'core/editor' ).getCurrentPostId(),
+		} ),
+		[]
+	);
 
-	useEffect( () => {
-		if ( globalStylesInUse ) {
-			recordUpgradeNoticePreSaveShow();
-		}
-	}, [ globalStylesInUse ] );
-
-	if ( ! globalStylesInUse ) {
-		return null;
-	}
-
-	return (
-		<Notice
-			status="warning"
-			isDismissible={ false }
-			className="wpcom-global-styles-notice notice-margin"
-		>
-			{ createInterpolateElement(
-				__(
-					'Your changes include customized styles that will only be visible once you <a>upgrade to a Premium plan</a>.',
```

Comment:
Something just crossed my mind. I think this notice is still needed in Site Editor, in browse mode, when the user saves a Style variation from the "Styles" panel. 

When the user clicks on the "Save" button, we display a save notice where we also mention that they are about to save a Global Styles change and they have to upgrade.

Reply:
In that mode, from what I've observed, we don't display the Site Editor warning notice.

Reply:
Oh nice catch, I totally missed that. I'll add this notice back but only when on the "view" canvas mode.